nomos-da/full-replication/src/lib.rs

```diff
@@ -123,12 +124,31 @@ impl attestation::Attestation for Attestation {
     }
 }
 
-#[derive(Debug, Clone, Serialize, Deserialize)]
+#[derive(Debug, Clone, Serialize, Deserialize, Eq, PartialEq)]
 pub struct Certificate {
     attestations: Vec<Attestation>,
 }
 
-impl certificate::Certificate for Certificate {}
+impl Hash for Certificate {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        state.write(certificate::Certificate::as_bytes(self).as_ref());
+    }
+}
+
+impl certificate::Certificate for Certificate {
+    type Blob = Blob;
+
+    fn blob(&self) -> <Self::Blob as blob::Blob>::Hash {
+        self.attestations[0].blob
+    }
+
+    fn as_bytes(&self) -> Bytes {
```

Comment:
I supposed we should be able to deserialize from `as_bytes` but this is variable length without mentioning how many attestations we have, am I missing something? Also, wasn't as bytes supposed to be `wire::deserialize`?

Reply:
Yes, but I saw it was manually added in the `Attestation` and did the same. Will change both (?)